R/checks.R

```diff
@@ -132,3 +132,41 @@ checkOutput <- function(output, call = parent.frame()){
   }
 }
 
+#' @noRd
+checkStudyPeriod <- function(cdm, studyPeriod) {
+  if (!isTRUE(inherits(cdm, "cdm_reference"))) {
+    cli::cli_abort("cdm must be a cdm_reference object.")
+  }
+  studyPeriod <- as.character(studyPeriod)
+  assertCharacter(studyPeriod, length = 2, na = TRUE)
+  observationRange <- cdm$observation_period |>
+    dplyr::summarise(minobs = min(.data$observation_period_start_date),
+                     maxobs = max(.data$observation_period_end_date))
+
+  if(is.na(studyPeriod[1])){
+    studyPeriod[1] <- observationRange |>
+      dplyr::pull("minobs") |>
+      as.character()
+  } else {
+    if(observationRange |>
+       dplyr::pull("minobs") > studyPeriod[1]) {
+      cli::cli_alert(paste0("The observation period in the cdm starts in ",observationRange |>
```

Comment:
why you need this alert?

Reply:
Maybe it's not necessary. I just wondered if it would be informative, e.g., someone wants 1900 to 2010 but the cdm starts at 1950, they likely do not know it if they asked